lib/connectionContext.ts

```diff
@@ -75,17 +86,54 @@ export namespace ConnectionContext {
    */
   export const userAgent: string = "/js-event-hubs";
 
-  export function create(config: ConnectionConfig, tokenProvider?: TokenProvider): ConnectionContext {
+  function encodeMessage(body: any): any {
+    let result: any;
+    if (body != undefined)
+      if (isBuffer(body))
+        result = rhea.message.data_section(body);
+      else if (typeof body !== "string")
+        result = rhea.message.data_section(Buffer.from(JSON.stringify(body), "utf8"));
+      else
+        result = rhea.message.data_section(Buffer.from(body, "utf8"));
+    else
+      result = rhea.message.data_section(Buffer.from("", "utf8"));
+
+    return result;
+  }
+
+  function decodeMessage(body: any): any {
+    let processedBody: any = body;
+    if (isBuffer(body)) {
+      processedBody = body.toString("utf8");
```

Comment:
if someone provides a buffer which is not a sequence of utf8 bytes, this will give them a weird output. the string vs. buffer case is a tough one. i'd probably just `JSON.stringify()` it even if it's a string so that you can `JSON.parse()` it on the other side and it will come out as a string again and you'll only get a buffer if you gave a buffer

Reply:
I get the idea but a little confused. Just to be sure. In the end we don't want to stringify the buffer and return it as-is if they actually sent a buffer. Otherwise JSON.parse() should take care of things.